rules/expiring-todo-comments.js

```diff
@@ -0,0 +1,357 @@
+
+			const isInclusion = ['in', 'out'].includes(dependency.condition);
+			if (isInclusion) {
+				const [trigger, messageId] = dependency.condition === 'in' ?
+					[hasTargetPackage, 'havePackage'] :
+					[!hasTargetPackage, 'dontHavePackage'];
+
+				if (trigger) {
+					context.report({
+						node: null,
+						loc: comment.loc,
+						messageId,
+						data: {
+							package: dependency.name
+						}
+					});
+				}
+
+				continue;
+			}
+
+			const todoVersion = tryToCoerceVersion(dependency.version);
+			const targetPackageVersion = tryToCoerceVersion(targetPackageRawVersion);
+
+			if (!hasTargetPackage || !targetPackageVersion) {
+				// Can't compare ¯\_(ツ)_/¯
+				continue;
+			}
+
+			const compare = semverComparisonForOperator(dependency.condition);
+
+			if (compare(targetPackageVersion, todoVersion)) {
+				context.report({
+					node: null,
+					loc: comment.loc,
+					messageId: 'versionMatches',
+					data: {
+						comparison: `${dependency.name} ${dependency.condition} ${dependency.version}`
+					}
+				});
+			}
+		}
+
+		const pkgEngines = pkg.engines || {};
+
+		for (const engine of engines) {
+			uses++;
+			const targetPackageRawEngineVersion = pkgEngines[engine.name];
+			const hasTargetEngine = Boolean(targetPackageRawEngineVersion);
+
+			if (!hasTargetEngine) {
+				continue;
+			}
+
+			const todoEngine = tryToCoerceVersion(engine.version);
+			const targetPackageEngineVersion = tryToCoerceVersion(targetPackageRawEngineVersion);
+
+			const compare = semverComparisonForOperator(engine.condition);
+
+			if (compare(targetPackageEngineVersion, todoEngine)) {
+				context.report({
+					node: null,
+					loc: comment.loc,
+					messageId: 'engineMatches',
+					data: {
+						comparison: `${engine.name} ${engine.condition} ${engine.version}`
+					}
+				});
+			}
+		}
+
+		return uses === 0;
+	}
+
+	return {
+		Program() {
+			rules.Program(); // eslint-disable-line new-cap
+		}
+	};
+};
+
+const schema = [{
+	type: 'object',
+	properties: {
+		terms: {
+			type: 'array',
+			items: {
+				type: 'string'
+			}
+		},
+
+		ignoreDatesOnPR: {
+			type: 'boolean'
+		}
+	}
+}];
+
+module.exports = {
+	create,
+	meta: {
+		type: 'suggestion',
+		docs: {
+			url: getDocsUrl(__filename)
+		},
+		messages: {
+			avoidMultipleDates: 'Avoid using multiple expiration dates for TODO {{ expirationDates }}',
+			expiredTodo: 'You have a TODO that past due date {{ expirationDate }}',
+			reachedPackageVersion: 'You have a TODO that past due package version {{ comparison }}',
+			avoidMultiplePackageVersions: 'Avoid asking multiple package versions for TODO {{ versions }}',
+			havePackage: 'You have a TODO that is deprecated since you installed {{ package }}',
+			dontHavePackage: 'You have a TODO that is deprecated since you uninstalled {{ package }}',
+			versionMatches: 'You have a TODO match for version for package {{ comparison }}',
+			engineMatches: 'You have a TODO match for engine version {{ comparison }}'
+		},
```

Comment:
A few phrasing tweaks:

```
		messages: {
 			avoidMultipleDates: 'Avoid using multiple expiration dates in TODO: {{ expirationDates }}',
 			expiredTodo: 'There is a TODO that is past due date: {{ expirationDate }}',
 			reachedPackageVersion: 'There is a TODO that is past due package version: {{ comparison }}',
 			avoidMultiplePackageVersions: 'Avoid using multiple package versions in TODO: {{ versions }}',
 			havePackage: 'There is a TODO that is deprecated since you installed: {{ package }}',
 			dontHavePackage: 'There is a TODO that is deprecated since you uninstalled: {{ package }}',
 			versionMatches: 'There is a TODO match for package version: {{ comparison }}',
 			engineMatches: 'There is a TODO match for engine version: {{ comparison }}'
 		},
```

Reply:
Added.